api/src/main/java/org/apache/iceberg/UpdateTableStatistics.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/** API for updating statistics files in a table. */
+public interface UpdateTableStatistics extends PendingUpdate<Snapshot> {
```

Comment:
There's no need to add this here, I'd focus on `TableMetadata` first.

Reply:
same as in https://github.com/apache/iceberg/pull/5450#discussion_r939975745